src/mtev_conf.h

```diff
@@ -44,6 +44,26 @@
 
 typedef void * mtev_conf_section_t;
 
+typedef enum {
+  MTEV_PARAM_INT8,
+  MTEV_PARAM_UINT8,
+  MTEV_PARAM_INT16,
+  MTEV_PARAM_UINT16,
+  MTEV_PARAM_INT32,
+  MTEV_PARAM_UINT32,
+  MTEV_PARAM_INT64,
+  MTEV_PARAM_UINT64,
+  MTEV_PARAM_FLOAT,
+  MTEV_PARAM_DOUBLE,
+  MTEV_PARAM_BOOLEAN
+} mtev_param_type_t;
+
+typedef mtev_boolean (*mtev_param_validator_t)(mtev_param_type_t, void *mem);
+mtev_boolean
+mtev_conf_register_global_param(const char *name, const char *xpath,
+                                mtev_param_type_t ptype, void *mem,
+                                mtev_param_validator_t vf);
+
```

Comment:
I'd consider using a parser / display function, rather than a type and validator? e.g.,
```c
typedef int (*mtev_param_parser_t)(const struct _mtev_param *param, const char *arg);
typedef struct _mtev_param {
  const mtev_param_parser_t *parser;
  const char *help_str;
  void *data;
} mtev_param_t
```
? This _could_ be useful for allowing more semantic information in the parser, e.g., to be able to parse `1day12hr`, or perhaps enum variants, and keeps the validation with the parsing... mtev would then define standard `mtev_param_t`'s for things like booleans...

I also think it'd be neat for this work with the `mtev_confstr` facility for parsing.